src/vm/types/mod.rs

```diff
@@ -35,16 +35,72 @@ pub struct ListData {
     pub type_signature: ListTypeData
 }
 
-// a standard principal is a version byte + hash160 (20 bytes)
 #[derive(Debug, Clone, Eq, PartialEq, Hash, Serialize, Deserialize)]
 pub struct StandardPrincipalData(pub u8, pub [u8; 20]);
 
+impl StandardPrincipalData {
+
+    pub fn transient() -> Self {
+        Self(1, [1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1])
+    }
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, Hash, Serialize, Deserialize)]
+pub struct QualifiedContractIdentifier {
+    pub issuer: StandardPrincipalData,
+    pub name: ContractName
+}
+
+impl QualifiedContractIdentifier {
+
+    pub fn new(issuer: StandardPrincipalData, name: ContractName) -> Self {
```

Comment:
Not a bug, but stylistically, we have tended not to use `Self`.  Instead, we use the name of the struct directly.  `Self` is used typically only when defining traits.

Reply:
Addressed with https://github.com/blockstack/blockstack-core/pull/1105/commits/e793428eea48870e61ce8665e8d52c7077ab519a